packages/field-base/src/input-control-mixin.js

```diff
@@ -103,13 +103,25 @@ export const InputControlMixin = (superclass) =>
 
     /** @protected */
     get slotStyles() {
-      // Needed for Safari, where ::slotted(...)::placeholder does not work
       return [
         `
+          /* Needed for Safari, where ::slotted(...)::placeholder does not work */
           :is(input[slot='input'], textarea[slot='textarea'])::placeholder {
             font: inherit;
             color: inherit;
           }
+
+          /* Override built-in autofill styles */
+          input[slot='input']:autofill {
+            -webkit-text-fill-color: var(--vaadin-input-field-autofill-color, black);
+            transition: background-color 9999s step-end;
```

Comment:
What is the purpose of this transition?

Reply:
The browser’s internal styles always override the input background color when autofilled. This transition delays that color from becoming visible. See https://stackoverflow.com/questions/2781549/removing-input-background-colour-for-chrome-autocomplete for more examples.

Reply:
Replaced transition with background-clip.